examples/cpp/keyvaluestore/client.cc

```diff
@@ -77,9 +79,20 @@ int main(int argc, char** argv) {
   // are created. This channel models a connection to an endpoint (in this case,
   // localhost at port 50051). We indicate that the channel isn't authenticated
   // (use of InsecureChannelCredentials()).
-  KeyValueStoreClient client(grpc::CreateChannel(
-      "localhost:50051", grpc::InsecureChannelCredentials()));
-  std::vector<std::string> keys = {"key1", "key2", "key3", "key4", "key5"};
+  // In this example, we are using a cache which has been added in as an
+  // interceptor.
+  grpc::ChannelArguments args;
+  std::vector<
+      std::unique_ptr<grpc::experimental::ClientInterceptorFactoryInterface>>
+      interceptor_creators;
+  interceptor_creators.push_back(std::unique_ptr<CachingInterceptorFactory>(
+      new CachingInterceptorFactory()));
```

Comment:
After this example, we may want another example with the interaction between two interceptors and a hijack. The second one could be your dummy interceptor, like the test.